File: api/src/test/java/net/blueshell/api/factory/dto/response/RedirectResponseDTOFactory.java
package net.blueshell.api.factory.dto.response;

import lombok.RequiredArgsConstructor;
import net.blueshell.api.dto.response.RedirectResponseDTO;
import net.blueshell.api.factory.dto.BaseDtoFactory;
import org.springframework.stereotype.Component;

/**
 * Factory for RedirectResponseDTO test instances.
 */
@Component
@RequiredArgsConstructor
public class RedirectResponseDTOFactory extends BaseDtoFactory<RedirectResponseDTO> {

    @Override
    public Class<RedirectResponseDTO> targetType() {
        return RedirectResponseDTO.class;
    }

    @Override
    public RedirectResponseDTO createBasic() {
        return new RedirectResponseDTO("/test/" + nextId());
    }
}

File: api/src/test/kotlin/net/blueshell/api/factory/dto/response/RedirectResponseDTOFactory.kt
package net.blueshell.api.factory.dto.response

import net.blueshell.api.factory.dto.BaseDtoFactory
import net.blueshell.api.domain.telemetry.web.dto.RedirectResponseDTO
import org.springframework.stereotype.Component

/**
 * Factory for RedirectResponseDTO test instances.
 */
@Component
class RedirectResponseDTOFactory : BaseDtoFactory<RedirectResponseDTO>() {

    override fun targetType(): Class<RedirectResponseDTO> = RedirectResponseDTO::class.java

    override fun createBasic(): RedirectResponseDTO = RedirectResponseDTO("/test/${nextId()}")
}
